composer/trainer/trainer_hparams.py

```diff
@@ -423,6 +448,11 @@ def initialize_object(self) -> Trainer:
 
         # ensure grad_accum is 'auto' or an integer
         grad_accum = _parse_grad_accum(self.grad_accum)
+        eval_batch_size = _parse_eval_batch_size(self.eval_batch_size)
+        # _parse_eval_batch_size ensures eval_batch_size is auto if set to str. We need to check if
+        # it's a str and not eval_batch_size == 'auto' or pyright complains it might be a str still.
+        if isinstance(eval_batch_size, str):
+            eval_batch_size = self.train_batch_size
```

Comment:
wouldn't this then just ignore `eval_batch_size=auto` completely? Or is that fact that auto was selected recorded somewhere else?

Reply:
if grad_accum=auto, we turn it on for both train/eval (per research's feedback). So, whether you set eval_batch_size explicitly or not, its enabled with grad_accum=auto. 

As a further feature, if you don't want to worry about eval batch size at all (most use cases), setting it to auto lets you ignore it completely.